source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md

```diff
@@ -0,0 +1,155 @@
+
+Instead, README goes right to the point:
+ 
+- **YES and NO code snippets**,
+- **how to use them** - copy/paste CLI command 
+- and **how to configure them** - link to particular lines in `ruleset.xml` and `easy-coding-standard.neon`.
+   
+<img src="/assets/images/posts/2017/object-calisthenics/rule6.png" class="thumbnail">
+
+### Configure What You Need
+
+As you can see in the bottom part of screenshot, most of rules are configurable. It allows you **to adapt their strictness to your specific needs and needs of your project**.
+ 
+*Do you prefer to require min 4 chars?*
+
+Configure **in CodeSniffer:**
+
+```xml
+<!-- ruleset.xml -->
+
+<?xml version="1.0"?>
+<ruleset name="ObjectCalisthenics">
+    <!-- Rule 6: Do not abbreviate -->
+    <rule ref="ObjectCalisthenics.NamingConventions.ElementNameMinimalLength">
+        <properties>
+            <property name="minLength" value="4"/> <!-- default: 3 -->
+        </properties>
+    </rule>
+</ruleset>
+```
+
+Configure **in [EasyCodingStandard](https://github.com/Symplify/EasyCodingStandard):**
+
+```yaml
+# easy-coding-standard.neon
+
+checkers:
+    # Rule 6: Do not abbreviate
+    ObjectCalisthenics\Sniffs\NamingConventions\ElementNameMinimalLengthSniff:
+        minLength: 4 # default: 3
+```
+
+*Do you want to add "y" to allowed short names?*
+
+Configure **in CodeSniffer:**
+
+```xml
+<!-- ruleset.xml -->
+
+<?xml version="1.0"?>
+<ruleset name="ObjectCalisthenics">
+    <!-- Rule 6: Do not abbreviate -->
+    <rule ref="ObjectCalisthenics.NamingConventions.ElementNameMinimalLength">
+        <properties>
+            <property name="minLength" value="4"/> 
+            <property name="allowedShortNames" type="array" value="y,i,id,to,up"/>
+            <!-- default: i,id,to,up -->
+        </properties>
+    </rule>
+</ruleset>
+```
+
+Configure **in [EasyCodingStandard](https://github.com/Symplify/EasyCodingStandard):**
+
+```yaml
+# easy-coding-standard.neon
+
+checkers:
+    # Rule 6: Do not abbreviate
+    ObjectCalisthenics\Sniffs\NamingConventions\ElementNameMinimalLengthSniff:
+        minLength: 4
+        allowedShortNames: ["y", "i", "id", "to", "up"] 
+        # default: ["i", "id", "to", "up"]
+```
+
+### Minitip: What Can You Configure in Particular Sniff?
+
+- Open `ElementNameMinimalLengthSniff` class in your IDE.
+- **Look for public properties**. CodeSniffer uses them to set any configuration. 
+
+
+## Thanks to all Contributors
+
+Last but not least, I'd like to personally thank contributors who helped to make this version happen as it is:
+
+- [@frenck](https://github.com/frenck)
+- [@mihaeu](https://github.com/mihaeu)
+- [@roukmoute](https://github.com/roukmoute)
+- [@UFOMelkor](https://github.com/UFOMelkor)
+
+Without your help, this would not have been possible.
+Thank you guys.
+
+
+## To Sum up Object Calisthenics 3.0
+
+- This release focuses **on simple setup** rather than on academic explanation of their meaning.  
```

Comment:
if you want to make it shorter and more memorable consider phrasing it like the agile manifesto:

simple setup over boring academic explanations